wezterm-gui/src/glyphcache.rs

```diff
@@ -733,16 +784,88 @@ impl<T: Texture2d> GlyphCache<T> {
         buffer.log_bits();
         */
 
-        let sprite = self.atlas.allocate(&buffer)?;
-        self.block_glyphs.insert(block, sprite.clone());
-        Ok(sprite)
+        self.atlas.allocate(&buffer).map_err(Into::into)
     }
 
-    pub fn cached_block(&mut self, block: BlockKey) -> anyhow::Result<Sprite<T>> {
-        if let Some(s) = self.block_glyphs.get(&block) {
+    pub fn cached_custom_glyph(
+        &mut self,
+        custom_glyph: CustomGlyphKey,
+    ) -> anyhow::Result<Sprite<T>> {
+        if let Some(s) = self.custom_glyphs.get(&custom_glyph) {
             return Ok(s.clone());
         }
-        self.block_sprite(block)
+
+        let sprite = match custom_glyph {
+            CustomGlyphKey::Block(block) => self.block_sprite(block)?,
+            CustomGlyphKey::BoxDrawing(box_drawing) => self.box_drawing_sprite(box_drawing)?,
+        };
+
+        self.custom_glyphs.insert(custom_glyph, sprite.clone());
+        Ok(sprite)
+    }
+
+    fn box_drawing_sprite(&mut self, box_drawing: BoxDrawingKey) -> anyhow::Result<Sprite<T>> {
+        let mut buffer = Image::new(
+            self.metrics.cell_size.width as usize,
+            self.metrics.cell_size.height as usize,
+        );
+        let black = SrgbaPixel::rgba(0, 0, 0, 0);
+        let white = SrgbaPixel::rgba(0xff, 0xff, 0xff, 0xff);
+
+        let cell_rect = Rect::new(Point::new(0, 0), self.metrics.cell_size);
+
+        buffer.clear_rect(cell_rect, black);
+
+        let draw_rect = |buffer: &mut Image, rect: Box2D<usize, PixelUnit>| {
+            for x in rect.min.x..rect.max.x {
+                for y in rect.min.y..rect.max.y {
+                    let pixel = buffer.pixel_mut(x, y);
+                    *pixel = white.as_srgba32();
+                }
+            }
+        };
+
+        let center = cell_rect.center();
+        let light_thickness = self.metrics.underline_height as usize;
```

Comment:
I think that it may make sense to aspect correct this; most monospace fonts are taller than they are wide, so a height metric should probably be multiplied by `width / height` in floating point to convert to a width metric, and then have a min size of 1 pixel.  Eg: `(underline_height * width / height).min(1)`.  For the double thickness, multiply by 2 before converting to integer

Reply:
> most monospace fonts are taller than they are wide

That's true, but are you convinced that applies in this case? The intent of these characters is to _draw boxes_ on the screen, it would seem weird to me to have a vertical line be wider than a horizontal one, just because its length is also taller?

That is to say, if I were to draw a box using:

```
┌─┐
│ │
└─┘
```

I'd expect the width of the line to not differ between the vertical and horizontal lines (it would also mean the corners would have to draw their horizontal and vertical parts differently, which makes them not conform to the Unicode standard).

Unless your point is to have _all_ lines be drawn using the aspect ratio, which we can do, but then a horizontal line will be less thick than an underline, not sure if that matters?

Reply:
(Unless drawing actual underlines also already takes the aspect ratio in account?)

Reply:
if the font specifies that the underline height is 2px, and the cell metrics are 8px wide and 16px tall: if you make the vertical lines 2px wide then they will appear too wide.  Aspect correcting them will make them have the correct proportions, which in this case should be 1px wide (`2px * 8px / 16px`).

If you set the font size to something large (eg: 144 or even double that) then this difference will be easier to see.